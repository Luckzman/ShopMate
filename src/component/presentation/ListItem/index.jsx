import React from 'react';
import PropTypes from 'prop-types';
import ListGroup from 'react-bootstrap/ListGroup';

/**
 * @description This component render List Item of an array
 * 
 * @param {array} listArr
 * @param {JSX}
 */
const ListItem = ({ listArr, selectedItem }) => {
  
  return (
    <ListGroup variant="flush">
      {
        listArr.map((item, index) => {
          // console.log(item, 'item')
          return (
            <ListGroup.Item key={`${item} ${index}`} action onClick={() => selectedItem(index+1, item)}>
              {item.name}
            </ListGroup.Item>
          )
        })
      }
    </ListGroup>
  )
}

ListItem.propTypes = {
  listArr: PropTypes.array,
  selectedItem: PropTypes.func.isRequired,
}
ListItem.defaultProps = {
  listArr: [],
}

export default ListItem;
