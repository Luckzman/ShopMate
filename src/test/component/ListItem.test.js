import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../../component/presentation/ListItem';

describe('ListItem Component', () => {
  it('should render correctly', () => {
    const props = {
      listArr: [],
    }
    const wrapper = shallow(<ListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
