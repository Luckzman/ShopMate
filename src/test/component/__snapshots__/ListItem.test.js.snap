// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Component should render correctly 1`] = `ShallowWrapper {}`;
